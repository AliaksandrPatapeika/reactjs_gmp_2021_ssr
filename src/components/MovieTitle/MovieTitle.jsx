import './MovieTitle.less';

import PropTypes from 'prop-types';
import React from 'react';

const MovieTitle = ({title}) => <div className="movieTitle">{title}</div>;

MovieTitle.propTypes = {
  title: PropTypes.string.isRequired
};

export default MovieTitle;
