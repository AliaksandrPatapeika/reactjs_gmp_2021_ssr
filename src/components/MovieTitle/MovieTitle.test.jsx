import {head} from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

import MockMovies from '../../tests/mocks/mockData';
import MovieTitle from './MovieTitle';

describe('MovieTitle', () => {
  test('renders correctly', () => {
    const mockMovie = head(MockMovies);
    const movieTitle = renderer
      .create(<MovieTitle title={mockMovie.title} />)
      .toJSON();

    expect(movieTitle).toMatchSnapshot();
  });
});
