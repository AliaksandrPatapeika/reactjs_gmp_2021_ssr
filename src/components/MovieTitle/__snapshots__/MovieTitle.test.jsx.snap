// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTitle renders correctly 1`] = `
<div
  className="movieTitle"
>
  Pulp Fiction
</div>
`;
